tests/types/test_fraction.py

```
@@ -68,11 +62,28 @@ def test_fraction_validate_json(json_str, expected):
+    'str,expected',
+    [
+        ('1/3', Fraction(1, 3)),
+        ('42.5', Fraction('42.5')),
+        ('0', Fraction(0)),
+        ('42', Fraction(42)),
+        ('42.5', Fraction('42.5')),
+        ('0/100', Fraction(0)),
+        ('-47e-2', Fraction(-47, 100)),
+    ],
+)
+def test_fraction_validate_strings(str, expected):
```

Ruff (0.15.18)

[error] Function argument `str` is shadowing a Python builtin

(A002)